signhandler/src/main/java/se/swedenconnect/signservice/signature/tbsdata/impl/PDFTBSDataProcessor.java

```diff
@@ -141,9 +141,12 @@ protected void checkToBeSignedData(@Nonnull final byte[] tbsData, final boolean
             throw new InvalidRequestException("Signing time is not allowed in PAdES requests in strict processing");
           }
         }
+        // Test if signing time is current
+        this.checkSigningTime(Objects.requireNonNull(getCmsSigningTime(signedAttributes),
```

Comment:
Compact code perhaps isn't the best. This is unreadable. 

Also Objects.requireNonNull will throw a NPE, not an IllegalArgumentException.

I suggest you test if it is null explicitly and then throw a InvalidRequestException.

Reply:
Fixed this by fixing the underlying function to extract signing time from signed attributes